lint: Call more checks from test_runner

maflcko opened the pull request:
The lint `commit-script-check.sh` can not be called from the test_runner at all and must be called manually. Also, some checks require `COMMIT_RANGE` to be set.

Fix all issues by moving two lint checks into the test_runner. Also, the proper commit range is passed to the checks by the test_runner, so that the user no longer has to do it.

Commit:
lint: Move commit range printing to test_runner

Having a single test_runner for all logic improves the consistency and
UX.

Commit:
lint: Call lint_commit_msg from test_runner

Allowing to call the check from the test_runner allows for consistent
error messages. Also, manually setting the commit range is no longer
needed.

kevkevinpal commented on test/lint/test_runner/src/main.rs:
```diff
@@ -650,6 +724,10 @@ fn main() -> ExitCode {
     };
 
     let git_root = get_git_root();
+    let commit_range = commit_range();
+    let commit_log = check_output(git().args(["log", "--no-merges", "--oneline", &commit_range]))
+        .expect("git error");
+    println!("Checking commit range ({commit_range}):\n{commit_log}\n");
```
looks like this adds a new log added to the CI (which I think is fine)

For other reviewers this is what this [log looks like in Cirrus CI](https://cirrus-ci.com/task/4637065368829952?logs=lint#L186-L188)
```
[12:23:15.442] Checking commit range (HEAD~..HEAD):
[12:23:15.442] fa42b4aa5e lint: Call lint_commit_msg from test_runner
[12:23:15.442] faee335285 lint: Call lint_scripted_diff from test_runner
```

kevkevinpal commented on test/lint/test_runner/src/main.rs:
```diff
@@ -650,6 +724,10 @@ fn main() -> ExitCode {
     let git_root = get_git_root();
+    let commit_range = commit_range();
+    let commit_log = check_output(git().args(["log", "--no-merges", "--oneline", &commit_range]))
+        .expect("git error");
```
nit: 
In `commit_range` there is also a `.expect` which has the text "git command failed"
it would make sense to have the same text for consistency, but feel free to ignore

kevkevinpal replied:
yea should be fine just thought I'd point it out, I think it is fine as is

kevkevinpal reviewed (approved):
crACK [fa42b4a](https://github.com/bitcoin/bitcoin/pull/31653/commits/fa42b4aa5ed41b556f9c71f710decba3f2cdc8e6)

I think this is a good change moves some of our python linting scripts into our rust lint runner which reduces the number of python files we need to maintain and does the same thing.

Looks like this change adds another log which lists the commits being linted and their oneline which I think is fine

kevkevinpal commented on ci/lint/06_script.sh:
```diff
@@ -14,18 +14,9 @@ if [ -n "$CIRRUS_PR" ]; then
-  # HEAD, the range will be empty.
-  COMMIT_RANGE="$( git rev-list --max-count=1 --merges HEAD )..HEAD"
 fi
 export COMMIT_RANGE
```
I think we can delete this line and then do the following on line 12
`export COMMIT_RANGE="HEAD~..HEAD"`

since now this export sometimes is empty

Commit:
lint: Call lint_scripted_diff from test_runner

Allowing to call the check from the test_runner allows for consistent
error messages and better UX by having a single test_runner for all
checks.

This requires the env var to be set for now. The next commit makes the
commit range optional.

kevkevinpal commented:
reACK [faf8fc5](https://github.com/bitcoin/bitcoin/pull/31653/commits/faf8fc5487d409eeff7b7b260eabb6929a7b7a5f)

willcl-ark reviewed (approved):
tACK faf8fc5487d409eeff7b7b260eabb6929a7b7a5f

Makes sense to move this into test_runner.

Worked correctly for me, although I did forget that we currently can't run the linter (and local CI) in a worktree, which I was using.

I'll try and open a PR to fix that shortly.
